### PR TITLE
chore: refactor settings class to functional component

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

This giant class was causing me all kinds of pains when i tried to implement the revoke phone number task in the sprint. Turning this into a functional component so that we can use hooks and reduce boilerplate.

No actual changes aside from adding an analytics event to track disabling analytics 😅

### Test plan

n/a

### Related issues

- Fixes RET-717 

### Backwards compatibility

Y

### CODE REVIEW

File: src/account/Settings.tsx [hunk 1]
```diff
@@ -411,143 +365,139 @@ export class Account extends React.Component<Props, State> {
...
-            {!numberVerified && (
-              <SettingsItemTextValue title={t('confirmNumber')} onPress={this.goToConfirmNumber} />
-            )}
-            <SettingsItemTextValue
-              title={t('languageSettings')}
-              testID="ChangeLanguage"
-              value={currentLanguage?.name ?? t('unknown')}
-              onPress={this.goToLanguageSetting}
-            />
+  const currentLanguage = useSelector(currentLanguageSelector)
```
Comment (dievazqu) [inline src/account/Settings.tsx:368]: nit: Move this to the top of the function where all the selectors are.

**Review by dievazqu — APPROVED** [review]
LGTM!
The tests are failing, but I think it's because you have to mock the current language Selector. And maybe update the snapshots?

File: src/account/Settings.tsx (hunk 1 above)
Comment (kathaypacific) [reply src/account/Settings.tsx]: ah, thank you!! i forgot about this

File: src/account/Settings.tsx [hunk 2]
```diff
@@ -55,222 +58,167 @@ import {
...
+  const { sessions: walletConnectV2Sessions } = useSelector(selectSessionsV2)
+
+  const { v1, v2 } = useSelector(walletConnectEnabledSelector)
+  const supportedBiometryType = useSelector(supportedBiometryTypeSelector)
+  const shouldShowRecoveryPhraseInSettings = useSelector(shouldShowRecoveryPhraseInSettingsSelector)
+  const hapticFeedbackEnabled = useSelector(hapticFeedbackEnabledSelector)
+  const decentralizedVerificationEnabled = useSelector(decentralizedVerificationEnabledSelector)
+  const currentLanguage = useSelector(currentLanguageSelector)
+
+  const walletConnectEnabled = v1 || v2
```
Comment (kathaypacific) [inline src/account/Settings.tsx:110]: this change resulted in one of the snapshots updating. previously, this was only looking at whether v1 was enabled. i can't remember why it was like this, but this flag shows / hides the connected applications. idk why we'd want to hide them if v1 is off but v2 is on. so i added it

File: src/account/Settings.tsx [hunk 3]
```diff
@@ -55,222 +58,167 @@ import {
...
+
+  const { sessions: walletConnectV1Sessions } = useSelector(selectSessionsV1)
+  const { sessions: walletConnectV2Sessions } = useSelector(selectSessionsV2)
+
+  const { v1, v2 } = useSelector(walletConnectEnabledSelector)
+  const supportedBiometryType = useSelector(supportedBiometryTypeSelector)
+  const shouldShowRecoveryPhraseInSettings = useSelector(shouldShowRecoveryPhraseInSettingsSelector)
+  const hapticFeedbackEnabled = useSelector(hapticFeedbackEnabledSelector)
+  const decentralizedVerificationEnabled = useSelector(decentralizedVerificationEnabledSelector)
+  const currentLanguage = useSelector(currentLanguageSelector)
```
Comment (kathaypacific) [inline src/account/Settings.tsx:108]: we were previously using the language from `i18n` directly to display the language, but it should really be from redux. we weren't mocking `i18n` properly in the tests before so "unknown" was the language in the snapshots. 

side note: this is why i dislike snapshot testing. some snapshots don't even make sense. but i couldn't find the energy to refactor the tests today

**Comment by MuckT** [comment]
@kathaypacific it looks like the confirm account removal modal isn't displaying after completing the backup quiz.

**Comment by kathaypacific** [comment]
> @kathaypacific it looks like the confirm account removal modal isn't displaying after completing the backup quiz.

@MuckT is there anything in this PR that could have caused that? i'm confused about what the flow should look like, and whether the behaviour already exists on main

**Pull request merged**